oximeter/collector/src/agent.rs

```diff
@@ -434,6 +488,60 @@ impl OximeterAgent {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct CollectionTaskSenderWrapper {
+    single_tx: mpsc::Sender<CollectionTaskOutput>,
+    cluster_tx: mpsc::Sender<CollectionTaskOutput>,
+}
+
+impl CollectionTaskSenderWrapper {
+    pub async fn send(
+        &self,
+        msg: CollectionTaskOutput,
+        log: &Logger,
+    ) -> anyhow::Result<()> {
+        let (result_single, result_cluster) = futures::future::join(
+            self.single_tx.send(msg.clone()),
+            self.cluster_tx.send(msg),
+        )
+        .await;
+
+        if let Err(e) = result_single {
+            error!(
+                log,
+                "failed to send value from the collection task to channel for single node: {e:?}"
+            );
+        };
+        if let Err(e) = result_cluster {
```

Comment:
If I'm understanding correctly, we always create the `cluster_tx`, right? That means we will log all these messages, even when we will never start a clustered database. Is that OK? It's "harmless", I'm really asking how hard it will be to filter out those log messages.

One idea that might help is splitting out the loggers for each kind of database, or somehow adding a key that lets us easily filter them. Right now, I think you'd have to do that by some string matching.

Reply:
It's unlikely that there will be many of these messages. They should only happen when one of the tasks is down and the receiver gets dropped. This should only happen on Nexus shutdown or a task crash. Since there's only likely to be one of them, and very infrequently, I'm not sure there needs to be more detail than is given here.